gems/sorbet-runtime/bench/constructor.rb

```diff
@@ -25,13 +40,43 @@ class Subdoc < T::Struct
       prop :prop10, T::Hash[String, Subdoc], default: {}
     end
 
+    class ExampleWithoutTypecheck < T::Struct
+      class Subdoc < T::Struct
+        prop :prop, String, checked: :never
+      end
+
+      prop :prop1, T.nilable(Integer), checked: :never
+      prop :prop2, Integer, default: 0, checked: :never
+      prop :prop3, Integer, checked: :never
+      prop :prop4, T::Array[Integer], checked: :never
+      prop :prop5, T::Array[Integer], default: [], checked: :never
+      prop :prop6, T::Hash[String, Integer], checked: :never
+      prop :prop7, T::Hash[String, Integer], default: {}, checked: :never
+      prop :prop8, T.nilable(Subdoc), checked: :never
+      prop :prop9, T::Array[Subdoc], default: [], checked: :never
+      prop :prop10, T::Hash[String, Subdoc], default: {}, checked: :never
+    end
+
     def self.run
       input = {
         prop3: 0,
         prop4: [],
         prop6: {},
       }.freeze
 
+      100_000.times do
+        ExamplePoro.new(input)
+      end
+
+      result = Benchmark.measure do
+        1_000_000.times do
+          ExamplePoro.new(input)
+        end
+      end
```

Comment:
I'd be interested to see what the benchmarks look like if they weren't hacking around limitations in the (MRI) Ruby VM, and using keyword arguments instead.

Reply:
Surprisingly, I tried an example using keyword args and they were actually slower than using the hash 🤔 
```
Plain Ruby (μs/iter):
  0.463951   0.002514   0.466465 (  0.468203)
Plain Ruby Kwargs (μs/iter):
  0.475856   0.002753   0.478609 (  0.480072)
```

where the Kwargs example was:
<details>
<summary>Script</summary>

```ruby
class ExamplePoroKwargs
  def initialize(prop1: nil, prop2: 0, prop3:, prop4:, prop5: [], prop6:, prop7: {}, prop8: nil, prop9: [], prop10: {})
    @prop1 = prop1
    @prop2 = prop2
    @prop3 = prop3
    @prop4 = prop4
    @prop5 = prop5
    @prop6 = prop6
    @prop7 = prop7
    @prop8 = prop8
    @prop9 = prop9
    @prop10 = prop10
  end
end

# ...

prop3 = 0
prop4 = []
prop6 = {}

100_000.times do
  ExamplePoroKwargs.new(prop3: prop3, prop4: prop4, prop6: prop6)
end

result = Benchmark.measure do
  1_000_000.times do
    ExamplePoroKwargs.new(prop3: prop3, prop4: prop4, prop6: prop6)
  end
end

puts "Plain Ruby Kwargs (μs/iter):"
puts result
```

</details>

Reply:
This is what I would have expected; the keyword args force the Ruby VM to allocate a hash on every call, plus putting the arguments into the hash, etc.

I think the common behavior inside Stripe is to use keyword arguments, rather than passing in a hash (though I guess some people might be getting JSON arguments off the wire and passing that hash into a `T::Struct` constructor directly?).